docker/docker_client.go

```diff
@@ -383,7 +399,7 @@ func (c *dockerClient) makeRequestToResolvedURL(ctx context.Context, method, url
 	if c.sys != nil && c.sys.DockerRegistryUserAgent != "" {
 		req.Header.Add("User-Agent", c.sys.DockerRegistryUserAgent)
 	}
-	if sendAuth {
+	if auth != noAuth {
```

Comment:
This silently uses v2 when `auth == v1Auth`. Can it fail instead? Or comment out the `v1Auth` enum value declaration so that no-one can accidentally use it and expect it to work.

Reply:
oh whoops. Changed it to `if auth == v2Auth` and commented out the v1Auth enum with a TODO above it.